reference/supported-languages.md

```diff
@@ -0,0 +1,36 @@
+# Languages/Formats OpenRewrite supports
+
+OpenRewrite is continuously adding support for new languages and formats. As of June 2023, OpenRewrite supports:
+
+## Programming languages
+
+* Java
+* Kotlin
+* Groovy
+* COBOL (open-source, but not free for commercial use)
```

Comment:
Is it too early to add Python & TS/JS? They're pretty close...
```suggestion
* COBOL (open-source, but not free for commercial use)
* Python
* TypeScript / JavaScript
```

Reply:
Should we include Python and TypeScript even though those are only sort of supported right now?

Reply:
I'm unsure -- I left them out because it felt like we didn't _really_ support them, yet. 

Tracey said: 

```
Python is just a simple text search for now; it’s not functional yet.
TypeScript is in the early stages of support.
```